apps/central-scan/backend/src/importer.ts

```diff
@@ -49,58 +45,37 @@ export class Importer {
   private readonly scanner: BatchScanner;
   private sheetGenerator?: BatchControl;
   private batchId?: string;
-  private workerPool?: WorkerPool<workers.Input, workers.Output>;
-  private readonly workerPoolProvider: () => WorkerPool<
-    workers.Input,
-    workers.Output
-  >;
-  private interpreterReady = true;
-
-  constructor({
-    workspace,
-    scanner,
-    workerPoolProvider = (): WorkerPool<workers.Input, workers.Output> =>
-      inlinePool<workers.Input, workers.Output>(workers.call),
-  }: Options) {
+  private interpreterState: 'init' | 'configuring' | 'ready' = 'init';
+  private interpreterReadyPromise?: Promise<void>;
```

Comment:
Do we need to store this promise? Seems like we could equivalently just check `interpreterState`

Reply:
We can't just check `interpreterState` because if configuration is in progress and `configureInterpreter` is called again it will just resolve immediately. Holding on to the promise ensures that all callers are resolved at the same time and only _after_ the interpreter is ready.

Reply:
Gotcha, thanks!